src/Donors/resources/admin/components/Notifications/Notification.tsx

```diff
@@ -0,0 +1,74 @@
+import {useEffect} from '@wordpress/element';
```

Comment:
Hey Paulo, why do we need to duplicate this? Do we have to do this for every admin screen? I don't think so. We should extract the shared logic so that it can be used on every screen. We don't have to reinvent the wheel each time we add a new screen.

Reply:
@alaca Refactored this and created a generic `AdminDetailsPage` component that hold the main shared logic while still allowing customizations when needed.

![CleanShot 2025-05-22 at 16 27 00](https://github.com/user-attachments/assets/a2ef83a2-e814-43d8-b2bf-a1f944848a19)